### ISSUE TITLE
[Bug]: LC0091 False-positive missing translation when property casing in source differs from canonical

### ISSUE THREAD

**Issue by MODUSCarstenScholling** [ALCops/Analyzers#411]
### Analyzer

LinterCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

LC0091

### Description

`TranslatableTextShouldBeTranslated` (LC0091) reports a false-positive missing translation when the developer writes a translatable property using non-canonical casing (e.g. `Tooltip` or `TOOLTIP` instead of `ToolTip`). The XLIFF file produced by the AL compiler contains the correct translation, but the analyzer computes a different trans-unit ID and therefore fails to find it.

XLIFF (compiler-generated, canonical casing):

```xml
<trans-unit id="Page 2931038265 - Control 1296262074 - Property 1295455071" ...>
  <source>This is a tooltip.</source>
  <target>Dies ist ein ToolTip.</target>
  <note ...>Page MyPage - Control MyField - Property ToolTip</note>
</trans-unit>
```

Analyzer output:

```
LC0091: Missing translation for 'Tooltip' in language(s): de-DE
```

Expected: no diagnostic — the translation exists in the XLIFF.

### Root cause

The trans-unit ID is composed of hashed segments (object, control, property). The AL compiler hashes the **canonical** property name (`ToolTip` → `1295455071`), while the analyzer hashes the **source-cased** name (`Tooltip` → `2001309823`). The IDs do not match, so the translation lookup fails.

The SDK's translation-ID methods (`LanguageFileUtilities.GetTranslationFileId` on newer SDKs, `GetLanguageSymbolId` / `GetLabelTextConstLanguageSymbolId` on older ones) derive the hashed segment from `symbol.Name` on the live `IPropertySymbol` instance. There is no public API to override the name, and the `name` string parameter on `GetTranslationFileId` only feeds the human-readable note segment — not the hash.

### Impact

- Applies to any translatable property (`Caption`, `ToolTip`, `PromotedActionCategories`, `InstructionalText`, `AboutTitle`, `AboutText`, `OptionCaption`, …) written with non-canonical casing.
- AL is case-insensitive, so the source compiles cleanly; only LC0091 reports a false positive.
- Affects both new-SDK (`GetTranslationFileId`) and old-SDK (`GetLanguageSymbolId`) code paths in the helper.

### AL code to reproduce

Source (`Tooltip` with lowercase `t`):

```al
page 50100 MyPage
{
    layout
    {
        area(Content)
        {
            field(MyField; MyField)
            {
                Tooltip = 'This is a tooltip.';
            }
        }
    }
}
```

**Comment by Arthurvdv** [comment]
Closing this issue now that `v1.1.0` has been released.
